src/Middleware/RequestBodyBufferMiddleware.php

```diff
@@ -23,10 +23,11 @@ final class RequestBodyBufferMiddleware
     public function __construct($sizeLimit = null)
     {
         if ($sizeLimit === null) {
-            $sizeLimit = \ini_get('post_max_size');
+            $iniSizeLimit = \ini_get('post_max_size');
+            $sizeLimit = $iniSizeLimit !== '0' ? $iniSizeLimit : null;
         }
```

Comment:
Right now, a zero value will only mean "unlimited" when not given explicitly and read from `post_max_size`. This makes it a bit inconsistent with passing `$sizeLimit` explicitly and also makes it harder to test. Should both paths be handled consistently here? Among others, this should make testing this feature much easier.